### ISSUE TITLE
`cpptools` high CPU usage when no directory opened

### ISSUE THREAD

**Issue by daniel-stoneuk** [microsoft/vscode-cpptools#9079]
Bug type: General

Discovered due to issue [1478 in vscode-arduino](https://github.com/microsoft/vscode-arduino/issues/1478).

**Describe the bug**
- OS and Version: MacOS 12.3 21E230
- VS Code Version: VSCode Insiders (Can also be reproduced in Release, using Insiders to isolate extensions)
- C/C++ Extension Version: v1.9.6 (Also in Release)
- Other extensions you installed (and if the issue persists after disabling them): None

If the C/C++ extension is activated when no directory opened, the extension appears to try and index every single file in the home directory. This uses 100% CPU usage and can continue for a very long time. I believe this is similar to the issue at #6149, though that issue does not seem to address no directory being open.

Unfortunately, the `cpptools` process that is spawned by the extension also continues to run as a orphan process even after the window and VSCode are closed (through standard close buttons).

**To Reproduce**
1. Open an empty VSCode window and install the C/C++ extension
3. Open ~/Downloads/test.cpp
4. Observe cpptools CPU activity

<img width="1541" alt="image" src="https://user-images.githubusercontent.com/15860949/159792708-e33fff6e-c4d8-4ab3-98b7-040866e71db2.png">

6. Close VSCode
7. Observe cpptools CPU activity

<img width="735" alt="image" src="https://user-images.githubusercontent.com/15860949/159792836-c948a267-8611-446b-bcdf-089e884f0263.png">

(parent process null)

~/Downloads/test.cpp:
```
hello
```

**Expected behavior**

The `cpptools` process is terminated with the VSCode process. Maybe a default exclusion list for similar $home directories, or don't process files when no folder is open.

**Comment by Colengms** [comment]
Hi @daniel-stoneuk .  I'm not able to easily repro this.

The only headers that should be indexed by cpptools when a folder is not opened would be the system include paths returned by the compiler.  Is it possible there has been a symlink added somewhere under your system include paths, perhaps pointing to a parent directory and causing recursion?

**Comment by daniel-stoneuk** [comment]
Thanks for checking this out!

Which OS have you tried to repro this? I've run a command (`ls -lrt  `find ./ -follow -type l``) to find circular symlinks, but this alone takes a very long time. Before I run it for longer than 5 minutes, I wanted to check if you're also using macOS.

I'm not seeing this behaviour when opening a C++ project folder, would this indicate that it is not related to the system include paths?

For some extra information, here are truncated logs in a project directory "projectdir"

```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/local/bin/gcc-11'
Compiler query command line: "/usr/local/bin/gcc-11" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
terminating child process: 4350
Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/local/bin/gcc-11'
Compiler query command line: "/usr/local/bin/gcc-11" -std=gnu++17 -Wp,-v -E -dM -x c++ -m64  /dev/null
terminating child process: 4352
  /Users/danielstone/University Third Year/projectdir/** is not a directory
  Folder: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/JavaRuntimeSupport.framework/Versions/A/Headers/ will be indexed
  Folder: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/CHIP.framework/Versions/A/Headers/ will be indexed
 ...
  Processing folder (recursive): /Users/danielstone/University Third Year/projectdir/
  Processing folder (recursive): /usr/local/Cellar/gcc/11.2.0_3/include/c++/11/
  Processing folder (recursive): /usr/local/Cellar/gcc/11.2.0_3/lib/gcc/11/gcc/x86_64-apple-darwin21/11/include/
  Processing folder (recursive): /usr/local/Cellar/gcc/11.2.0_3/lib/gcc/11/gcc/x86_64-apple-darwin21/11/include-fixed/
  Discovering files: 19412 file(s) processed
  11013 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
  ...
```

And here are the logs with no folder open, and a C++ file activating the extension:

```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/local/bin/gcc-11'
Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/local/bin/gcc-11'
```

**Comment by Colengms** [comment]
Hi @daniel-stoneuk .  It doesn't look like it's getting to the point of parsing headers.  Could you try attaching lldb (or gdb) to the cpptools process, pausing the process when it's busy like this, and getting a backtrace?  The stack might help us understand what's going wrong.

One uncommon aspect of your scenario is use of gcc-11 on MacOS.  Given that the issue happens immediately after a log statement related to querying the compiler, the issue could be with parsing output from gcc-11 on Mac for some reason.  How did you install gcc-11?  Also, it's unclear to me why gcc-11 would be chosen as a default compiler on a Mac.  Is clang (which comes with XCode) not also installed?

Yes, I tried on an Intel Mac.  I also tried against on a M1 Mac.  :)  With 1.9.7.

**Comment by daniel-stoneuk** [comment]
Cheers @Colengms. 

The `gcc-11` is from [Homebrew](https://formulae.brew.sh/formula/gcc), so I have changed the default settings for the extension such that

```
"C_Cpp.default.compilerPath": "/usr/bin/gcc",
```

This does change the amount of logging that is outputted. I've attached the logs and stacktrace for no open directory. Additionally, while it was running I received a disk permission popup and Activity Monitor indicates that it is checking files in my home directory.

![Screenshot 2022-03-24 at 11 04 43](https://user-images.githubusercontent.com/15860949/159905050-ca44890c-cf26-4148-906d-e8712744c632.png)

![image](https://user-images.githubusercontent.com/15860949/159905353-1ab74872-eec8-4caa-b071-f8ed3c7688e8.png)

I'm now going to check for any symlinks within the system include directories returned by `/usr/bin/gcc -xc -E -v -`.

<details>
<summary>/usr/bin/gcc -xc -E -v -</summary>

```
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: x86_64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
 "/Library/Developer/CommandLineTools/usr/bin/clang" -cc1 -triple x86_64-apple-macosx12.0.0 -Wundef-prefix=TARGET_OS_ -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -E -disable-free -disable-llvm-verifier -discard-value-names -main-file-name - -mrelocation-model pic -pic-level 2 -mframe-pointer=all -fno-strict-return -fno-rounding-math -munwind-tables -target-sdk-version=12.3 -fvisibility-inlines-hidden-static-local-var -target-cpu penryn -tune-cpu generic -debugger-tuning=lldb -target-linker-version 762 -v -resource-dir /Library/Developer/CommandLineTools/usr/lib/clang/13.1.6 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Perl/5.28/darwin-thread-multi-2level/CORE -I. -I/usr/local/include -internal-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Library/Developer/CommandLineTools/usr/lib/clang/13.1.6/include -internal-externc-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Library/Developer/CommandLineTools/usr/include -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -Wno-reserved-identifier -Wno-gnu-folding-constant -Wno-objc-load-method -fdebug-compilation-dir=/Users/danielstone/testing-vscode -ferror-limit 19 -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcommon -fcolor-diagnostics -clang-vendor-feature=+messageToSelfInClassMethodIdReturnType -clang-vendor-feature=+disableInferNewAvailabilityFromInit -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -clang-vendor-feature=+enableAggressiveVLAFolding -clang-vendor-feature=+revert09abecef7bbf -clang-vendor-feature=+thisNoAlignAttr -clang-vendor-feature=+thisNoNullAttr -mllvm -disable-aligned-alloc-awareness=1 -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o - -x c -
clang -cc1 version 13.1.6 (clang-1316.0.21.2) default target x86_64-apple-darwin21.4.0
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Perl/5.28/darwin-thread-multi-2level/CORE"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 .
 /usr/local/include
 /Library/Developer/CommandLineTools/usr/lib/clang/13.1.6/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
 /Library/Developer/CommandLineTools/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
```
</details>


```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: "/usr/bin/gcc" -x c++ -E -dM /dev/null
terminating child process: 10203
Detected language standard version: c++98
Querying compiler for default C language standard using command line: "/usr/bin/gcc" -x c -E -dM /dev/null
terminating child process: 10205
Detected language standard version: c17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-apple-darwin21.4.0

terminating child process: 10207
Compiler query command line: "/usr/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64 -fno-blocks /dev/null
terminating child process: 10208
Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/bin/gcc'
Compiler query command line: "/usr/bin/gcc" -std=gnu++17 -Wp,-v -E -dM -x c++ -m64 -fno-blocks /dev/null
terminating child process: 10210
Database safe to open
```

<details>
<summary>The stacktrace:</summary>

```
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007ff8074390ea libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007ff807475c95 libsystem_pthread.dylib`_pthread_join + 358
    frame #2: 0x000000010ebc376a cpptools`msvc::thread_t::join() + 42
    frame #3: 0x000000010e34541c cpptools`vscode::message_handler::main_loop() + 4860
    frame #4: 0x000000010e340b1f cpptools`main + 287
    frame #5: 0x000000011c41551e dyld`start + 462
  thread #2
    frame #0: 0x00007ff80743c34e libsystem_kernel.dylib`kevent + 10
    frame #1: 0x000000010ebdcf2b cpptools`uv__io_poll + 859
    frame #2: 0x000000010ebd58a1 cpptools`uv_run + 465
    frame #3: 0x000000010eba4620 cpptools`msvc::loop_t::run_loop() + 64
    frame #4: 0x000000010eba851a cpptools`msvc::thread_t::invoker_t<void (*)(std::__1::shared_ptr<msvc::loop_t>), std::__1::shared_ptr<msvc::loop_t>&>::invoke() + 42
    frame #5: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #6: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #7: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #3
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #4
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #5
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #6
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #7
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #8
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #9
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #10
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #11
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #12
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #13
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #14
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2ca2 libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x000000010e449946 cpptools`msvc::thread_pool::do_work(unsigned long) + 406
    frame #4: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #5: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #6: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #15
    frame #0: 0x00007ff80743a3ea libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007ff807474a6f libsystem_pthread.dylib`_pthread_cond_wait + 1249
    frame #2: 0x00007ff8073d2d33 libc++.1.dylib`std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
    frame #3: 0x000000010e3273ec cpptools`std::__1::cv_status std::__1::condition_variable::wait_until<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > > const&) + 172
    frame #4: 0x000000010e44b667 cpptools`unsigned int msvc::bitset_event_t::wait_for_any_set<long long, std::__1::ratio<1l, 1000l> >(std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000l> > const&, unsigned int) + 103
    frame #5: 0x000000010e44b525 cpptools`msvc::thread_t::invoker_t<msvc::thread_pool::thread_pool()::$_0>::invoke() + 117
    frame #6: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #7: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #8: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #16
    frame #0: 0x00007ff80743c34e libsystem_kernel.dylib`kevent + 10
    frame #1: 0x000000010ebdcf2b cpptools`uv__io_poll + 859
    frame #2: 0x000000010ebd58a1 cpptools`uv_run + 465
    frame #3: 0x000000010eba4620 cpptools`msvc::loop_t::run_loop() + 64
    frame #4: 0x000000010eba851a cpptools`msvc::thread_t::invoker_t<void (*)(std::__1::shared_ptr<msvc::loop_t>), std::__1::shared_ptr<msvc::loop_t>&>::invoke() + 42
    frame #5: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #6: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #7: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
  thread #17
    frame #0: 0x00007ff80743a60e libsystem_kernel.dylib`fstatat$INODE64 + 10
    frame #1: 0x00007ff8073523f3 libsystem_c.dylib`fts_stat2 + 117
    frame #2: 0x00007ff80735848c libsystem_c.dylib`fts_build + 2244
    frame #3: 0x00007ff8073528f6 libsystem_c.dylib`fts_read$INODE64 + 889
    frame #4: 0x00007ff80737bcfc libsystem_c.dylib`both_ftw + 816
    frame #5: 0x000000010ebc254a cpptools`msvc::recursive_directory_iterator_utf8::enumerate_files_non_recursive(char const*) + 74
    frame #6: 0x000000010ebb28a2 cpptools`msvc::recursive_directory_iterator_utf8::recursive_directory_iterator_utf8(msvc::path_utf8 const&, bool, int) + 802
    frame #7: 0x000000010e454038 cpptools`browse_engine::add_mac_framework_directory(unsigned long, char const*, bool) + 104
    frame #8: 0x000000010e4523e3 cpptools`browse_engine::add_directory(unsigned long, char const*, std::__1::vector<browse_engine::directory_info, std::__1::allocator<browse_engine::directory_info> >&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, bool, bool) + 227
    frame #9: 0x000000010e4522d2 cpptools`browse_engine::add_directory(unsigned long, char const*, bool) + 98
    frame #10: 0x000000010e37de60 cpptools`vscode::message_handler::set_browse_engine_directories(std::__1::shared_ptr<workspace_folder_settings> const&) + 208
    frame #11: 0x000000010e35d975 cpptools`vscode::message_handler::cpptools_didChangeCppProperties(vscode::C_Cpp_ConfigurationParams const&) + 5221
    frame #12: 0x000000010e349a6d cpptools`vscode::message_handler::dispatch(vscode::vscode_client_message&&, vscode::vscode_server_message&, vscode::message_handler::msg_proc_thread_token) + 10733
    frame #13: 0x000000010e346f97 cpptools`vscode::message_handler::handle_message(vscode::vscode_client_message&&, vscode::message_handler::msg_proc_thread_token) + 55
    frame #14: 0x000000010e3b7910 cpptools`msvc::thread_t::invoker_t<vscode::message_handler::main_loop()::$_0>::invoke() + 2176
    frame #15: 0x000000010ebc371f cpptools`msvc::thread_helper_t::thread_entry(void*) + 15
    frame #16: 0x00007ff8074744e1 libsystem_pthread.dylib`_pthread_start + 125
    frame #17: 0x00007ff80746ff6b libsystem_pthread.dylib`thread_start + 15
```
</details>

**Comment by daniel-stoneuk** [comment]
The `.` in the default includes looks like it might be the issue as I've tested on other machines (non macOS) and don't see this. 

It looks like this is a configuration issue on my machine and is unrelated to the extension, I'm really sorry about this. Thanks so much for your help.

Should the `cpptools` process not terminating when VSCode is closed be considered a separate issue?

**Comment by daniel-stoneuk** [comment]
A while back I had added the following line to my `.zshrc` to get past an error with a header file not being included. (https://stackoverflow.com/a/65442835). I'll be sure to check for side-effects for any small change like this in the future.

```
export CPATH=/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Perl/5.28/darwin-thread-multi-2level/CORE:$CPATH
```



I imagine that since updating to 13.3, the `MacOSX11.1.sdk` was removed from my system so it wasn't immediately obvious in the system include directories, and the `:` added an empty entry that made C search the cwd.
